wn/lmf.py

```diff
@@ -0,0 +1,639 @@
+        event, elem = next(events)
+
+    _assert_closed(event, elem, 'LexicalEntry')
+
+    return LexicalEntry(
+        attrs['id'],
+        lemma,
+        forms=tuple(forms),
+        senses=tuple(senses),
+        syntactic_behaviors=tuple(syntactic_behaviors),
+        status=attrs.get('status'),
+        note=attrs.get('note'),
+        confidence=_get_confidence(attrs),
+        dcmeta=_get_dublin_core_metadata(attrs),
+    )
+
+
+def _load_lemma(events) -> Lemma:
+    event, elem = next(events)
+    if event != 'start' or elem.tag != 'Lemma':
+        raise LMFError('expected a Lemma element')
+    attrs = elem.attrib
+    return Lemma(
+        attrs['writtenForm'],
+        _get_literal(attrs['partOfSpeech'], _pos),
+        script=attrs.get('script'),
+        tags=_load_tags_until(events, 'Lemma'))
+
+
+def _load_form(local_root, events) -> Form:
+    attrs = local_root.attrib
+    return Form(
+        attrs['writtenForm'],
+        script=attrs.get('script'),
+        tags=_load_tags_until(events, 'Form'))
+
+
+def _load_tags_until(events, terminus) -> Tuple[Tag, ...]:
+    event, elem = next(events)
+
+    tags: List[Tag] = []
+    while event == 'start' and elem.tag == 'Tag':
+        event, elem = next(events)
+        _assert_closed(event, elem, 'Tag')
+        tags.append(Tag(elem.text, elem.attrib['category']))
+        event, elem = next(events)
+
+    _assert_closed(event, elem, terminus)
+
+    return tuple(tags)
+
+
+def _load_sense(local_root, events) -> Sense:
+    attrs = local_root.attrib
+    event, elem = next(events)
+
+    relations: List[SenseRelation] = []
+    while event == 'start' and elem.tag == 'SenseRelation':
+        event, elem = next(events)
+        _assert_closed(event, elem, 'SenseRelation')
+        relations.append(_load_relation(elem, SenseRelation, _sense_rels))
+        event, elem = next(events)
+
+    examples: List[Example] = []
+    while event == 'start' and elem.tag == 'Example':
+        event, elem = next(events)
+        _assert_closed(event, elem, 'Example')
+        examples.append(_load_example(elem))
+        event, elem = next(events)
+
+    counts: List[Count] = []
+    while event == 'start' and elem.tag == 'Count':
+        event, elem = next(events)
+        _assert_closed(event, elem, 'Count')
+        counts.append(_load_count(elem))
+        event, elem = next(events)
+
+    _assert_closed(event, elem, 'Sense')
+
+    return Sense(
+        attrs['id'],
+        attrs['synset'],
+        relations=tuple(relations),
+        examples=tuple(examples),
+        counts=tuple(counts),
+        lexicalized=_get_bool(attrs.get('lexicalized', 'true')),
+        adjposition=_get_optional_literal(attrs.get('adjposition'), _adjpositions),
+        status=attrs.get('status'),
+        note=attrs.get('note'),
+        confidence=_get_confidence(attrs),
+        dcmeta=_get_dublin_core_metadata(attrs),
+    )
+
+
+_R = TypeVar('_R', SynsetRelation, SenseRelation)
+
```

Comment:
What's `_R`?

Reply:
It's a [type variable](https://docs.python.org/3/library/typing.html#typing.TypeVar) so that I can create a generic function to return either SynsetRelation or SenseRelation objects while still allowing it to type-check properly. It has no purpose other than type-checking.